code/model/SubgoalPolicy.cpp

```diff
@@ -1752,6 +1752,26 @@ void SubgoalPolicy::SampleSubgoalSequence (const Problem& _rProblem,
 		pSubgoal->p_PddlSubgoalPredicate
 			= vec_CandidatePredicates [pSubgoal->i_SubgoalSelection];
 
+    //TODO: Add Config Check to make sure this is valid. Else if question found and
+    // config, throw an error
+    if (0 == pSubgoal->p_PddlSubgoalPredicate->s_Name.compare("question")) {
+      pSubgoal->b_isQuestion = true;
+      String_dq_t dq_QuestionArgs;
+      //Parse Question from PddlString
+      String s_PredicateString = pSubgoal->p_PddlSubgoalPredicate->GetPddlString();
+      size_t i_Start = s_PredicateString.rfind("(") + 1;
+      size_t i_End = s_PredicateString.find(")");
+      String s_QuestionString = s_PredicateString.substr(i_Start, i_End - i_Start);
+      s_QuestionString.Split(dq_QuestionArgs, ' ');
+      //Parse Question type and query from question
+      String s_QuestionType = dq_QuestionArgs[1];
+      size_t i_QueryIndex = s_QuestionString.find(dq_QuestionArgs[2]);
```

Comment:
@nicola 

given a `valid_predicate.dictionary` entry `INDEX|1|question object fruit` (as described in #3 )
The thing returned by PddlString is of the form 

`"(< (question object fruit ) ) > 0"`

`s_QuestionString` trims this to just `"question object fruit"` 
For this example,

`s_QuestionType = "object"`
`s_QuestionQuery = "fruit"`
In another example, `s_QuestionQuery` could equal `"have fruit furnace glass"`, achieving your calling convention

`i_QueryIndex` is just the index in the `s_QuestionString` to start the substring to find the query. 

At a high level, this directly interfaces the PDDL semantics with the IR semantics you spec'ed

I don't think descriptions this low level belong in comments, i find them distracting from the point, ie. `Parse Question from PddlString` and `Parse Question type and query from question` and totally redundant.